functions/go/set-namespace/transformer/namespace.go

```diff
@@ -98,45 +71,152 @@ func (p *NamespaceTransformer) Config(o *fn.KubeObject) error {
 	return nil
 }
 
+// Transform replace the existing Namespace object, namespace-scoped resources' `metadata.name` and
+// other namespace reference fields to the new value.
 func (p *NamespaceTransformer) Transform(objects []*fn.KubeObject) {
 	p.SetDependsOnMap(objects)
 	for _, o := range objects {
 		switch {
+		// Skip local resource which `kpt live apply` skips.
+		case o.IsLocalConfig():
+			continue
 		case o.IsGVK("v1", "Namespace"):
-			o.SetName(p.Namespace)
+			o.SetName(p.NewNamespace)
 		case o.IsGVK("apiextensions.k8s.io/v1", "CustomResourceDefinition"):
-			err := o.Set(p.Namespace, "spec", "conversion", "webhook", "clientConfig", "service", "namespace")
-			if err != nil {
-				p.Errors = append(p.Errors, err.Error())
-			}
+			p.updateNamespace(o, "spec", "conversion", "webhook", "clientConfig", "service", "namespace")
 		case o.IsGVK("apiregistration.k8s.io/v1", "APIService"):
-			err := o.Set(p.Namespace, "spec", "service", "namespace")
-			if err != nil {
-				p.Errors = append(p.Errors, err.Error())
-			}
+			p.updateNamespace(o, "spec", "service", "namespace")
 		case o.GetKind() == "ClusterRoleBinding" || o.GetKind() == "RoleBinding":
 			var subjects []*Subject
 			o.GetOrDie(&subjects, "subjects")
```

Comment:
Do we have GetSlice?  Because if we do then we don't need to treat this specially...

Reply:
Good point! yeah, added.